src/errors/checkLocalTxInputs.ts

```diff
@@ -17,52 +22,100 @@ enum LocalTxType {
  * @param assetIds
  * @param amounts
  */
-export const checkLocalTxInput = (
+export const checkLocalTxInput = async (
+	api: ApiPromise,
 	assetIds: string[],
 	amounts: string[],
 	specName: string,
-	registry: Registry
-): LocalTxType => {
-	const relayChainInfo = registry.currentRelayRegistry;
-	const systemChainId = getChainIdBySpecName(registry, specName);
-	const systemParachainInfo = relayChainInfo[systemChainId];
-
+	registry: Registry,
+	isForeignAssetsTransfer?: boolean
+): Promise<LocalTxType> => {
 	// Ensure the lengths in assetIds and amounts is correct
 	if (assetIds.length > 1 || amounts.length !== 1) {
 		throw new BaseError(
 			'Local transactions must have the `assetIds` input be a length of 1 or 0, and the `amounts` input be a length of 1'
 		);
 	}
 
-	/**
-	 * We assume when the assetId's input is empty that the native token is to be transferred.
-	 */
-	if (assetIds.length === 0) {
-		return LocalTxType.Balances;
-	}
-
-	let assetId = assetIds[0];
+	if (isForeignAssetsTransfer) {
+		if (assetIds.length === 0) {
+			throw new BaseError(
+				'Local foreignAsset transactions must have the `assetIds` input be a length of 1'
+			);
+		}
+		// check the foreignAssetsInfo to see if the given foreignAssetMatches
```

Comment:
The comment here is a bit confusing specifically with `foreignAssetMatches`

Reply:
ill update this to be more clear